bin/cli.js

```diff
@@ -32,6 +33,87 @@ program.command("install [firmware]")
     .option("-p, --port <port>", "Serial port board is attached to")
     .option("-f, --firmata [firmata]", "Install firmata version of firmware")
     .option("-i, --address <address>", "Specify I2C address, eg 0x67")
-    .action(interchange.install_firmware.bind(interchange));
+    .option("--interactive", "Interactive mode will prompt for input")
+    .action(function (firmware, opts) {
+
+        if (opts.interactive) {
+
+            interchange.get_ports(function (err, ports) {
+
+                var questions;
+
+                if (err) {
+                    console.error(err);
+                    return;
+                }
+
+                questions = [
+                    {
+                        type: "list",
+                        name: "firmware",
+                        message: "Choose a firmware",
+                        choices: interchange.firmwares.map(function (el) {
+                            return el.name
+                        })
+                    },
+                    {
+                        type: "confirm",
+                        name: "firmata",
+                        message: "Install firmata version?",
+                        default: false
+                    },
+                    {
+                        type: "input",
+                        name: "firmataType",
+                        message: "Firmata name [optional]",
+                        default: null,
+                        when: function (answers) {
+                            return answers.firmata;
+                        }
+                    },
+                    {
+                        type: "list",
+                        name: "avr",
+                        message: "Choose a board",
+                        choices: [
+                            "uno",
+                            "nano",
+                            "promini",
+                        ],
+                        default: "nano"
+                    },
+                    {
+                        type: "list",
+                        name: "port",
+                        message: "Choose a port",
+                        choices: ports.map(function (el) {
+                            return el.comName;
+                        }),
+                        default: null
+                    },
+                    {
+                        type: "input",
+                        name: "address",
+                        message: "Choose an I2C address [optional]",
+                        default: null
+                    }
```

Comment:
Depending on answer to the firmata question (ie you've installed the Firmata) then we probably should skip this question as it's not relevant.